static/js/components/dashboard/CourseAction.js

```diff
@@ -87,11 +90,15 @@ export default class CourseAction extends React.Component {
     }
   }
 
+  isFutureCourse = (run: CourseRun): boolean =>
+    R.not(isEnrollableRun(run)) || _.isEmpty(run.course_id)
+
   renderEnrollButton(run: CourseRun, actionType: string): React$Element<*> {
     return (
       <div className="course-action">
         <SpinnerButton
           className="dashboard-button enroll-button"
+          disabled={this.isFutureCourse(run)}
```

Comment:
it would be good to add a test to just make sure the `disabled` prop is set correctly in the right scenarios

Reply:
this is what i am doing and it is almost ready

Reply:
:+1:

Reply:
Done @aliceriot